File: haushaltsbuch/src/main/java/org/github/haschi/haushaltsbuch/api/_BeginneInventur.java
package org.github.haschi.haushaltsbuch.api;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import org.axonframework.commandhandling.TargetAggregateIdentifier;
import org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de.Aggregatkennung;
import org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de.Anweisung;
import org.immutables.value.Value;

@Anweisung
@JsonSerialize(as = BeginneInventur.class)
@JsonDeserialize(as = BeginneInventur.class)
public interface _BeginneInventur
{
    @TargetAggregateIdentifier
    @Value.Parameter
    Aggregatkennung id();
}

File: haushaltsbuch/src/main/kotlin/org/github/haschi/haushaltsbuch/api/BeginneInventur.kt
package org.github.haschi.haushaltsbuch.api

import org.axonframework.commandhandling.TargetAggregateIdentifier
import org.github.haschi.haushaltsbuch.infrastruktur.modellierung.de.Aggregatkennung

data class BeginneInventur(@TargetAggregateIdentifier val id: Aggregatkennung)
